sklearn/utils/fixes.py

```diff
@@ -332,3 +334,24 @@ def _object_dtype_isnan(X):
     from collections import Iterable as _Iterable  # noqa
     from collections import Mapping as _Mapping  # noqa
     from collections import Sized as _Sized  # noqa
+
+
+def _joblib_parallel_args(prefer=None, require=None):
+    """Set joblib.Parallel arguments in a compatible way for 0.11 and 0.12+"""
+    from . import _joblib
+    if prefer not in ['threads', 'processes', None]:
+        raise NotImplementedError('prefer=%s is not supported'
+                                  % prefer)
+    args = {}
+    if _joblib.__version__ >= LooseVersion('0.12.0'):
+        if prefer is not None:
+            args['prefer'] = prefer
+        if require is not None:
+            args['require'] = require
```

Comment:
Why not simply `args = {'prefer': prefer, 'require': require'}`?
The default value for both argument is `None` so this check is not useful.

To avoid having to re-writte this part if we add some arguments to `joblib.Parallel`, I would suggest a function such as:

```python
def _joblib_parallel_args(**kwargs):
    if _joblib.__version__ >= LooseVersion('0.12'):
        return kwargs
    
    # If version is earlier than 0.12, mock the prefer and request arguments
    args = {'backend': None}
    prefer = kwargs.get('prefer')
    if prefer not in ['threads', 'processes', None]:
        raise NotImplementedError('prefer=%s is not supported'
                                  % prefer)
    args['backend'] = {'threads': 'threading',
                       'processes': 'multiprocessing'
                       None: None}[prefer]

    if kwargs.get('sharedmem', False):
        args['backend'] = 'threading'
    return args
```